cmd/antrea-controller/controller.go

```diff
@@ -96,23 +101,64 @@ func run(o *Options) error {
 
 	go apiServer.GenericAPIServer.PrepareRun().Run(stopCh)
 
+	if o.config.EnablePrometheusMetrics {
+		go initializePrometheusMetrics(
+			o.config.EnablePrometheusGoMetrics,
+			o.config.EnablePrometheusProcessMetrics)
+	}
+
 	<-stopCh
 	klog.Info("Stopping Antrea controller")
 	return nil
 }
 
+// Initialize Prometheus metrics collection.
+func initializePrometheusMetrics(
+	enablePrometheusGoMetrics bool,
+	enablePrometheusProcessMetrics bool) {
+	hostname, err := os.Hostname()
```

Comment:
Question - is it better to use K8s Node name?

Reply:
Is the pod spec reachable from within the pod? I think that using K8s hostname is preferred but I only found some hackish way to achieve that via environment variables.

Reply:
I think Agent knows about the Node name. For Controller could we just use a fixed name like "antrea-controller"? If we need Pod name, it is also known to Controller.
@tnqn

Reply:
Indeed node name is fetched via an environment variable at the agent.
I'll do the same for the controller as well if needed, and add both node names - os and k8s, as two labels on the same gauge.
Using a constant "antrea-controller' might be confusing if anyone ever manages two clusters on the same Prometheus server.

Reply:
Node name can conflict across clusters too.
But anyway, for Agent Node name might be good, as Agent runs on every Node, and Node name is easy to remember. But for Controller, its Node is not fixed. Probably Pod name makes more sense, esp. if we want to keep the same name even when Controller gets rescheduled?

Reply:
I could add both the pod name and the node as two separate labels, and change the metric name accordingly (e.g antrea_controller_runtime_info).
Or just leave the node name. What do you think?

Reply:
I think the idea sounds good to me: 
> add both the pod name and the node as two separate labels, and change the metric name accordingly (e.g antrea_controller_runtime_info).

Reply:
Cool. I'll apply the same for the agent then.